internal/replacements/replace_failed_process_groups.go

```diff
@@ -45,7 +45,7 @@ func getMaxReplacements(cluster *fdbv1beta2.FoundationDBCluster, maxReplacements
 
 // ReplaceFailedProcessGroups flags failed processes groups for removal and returns an indicator
 // of whether any processes were thus flagged.
-func ReplaceFailedProcessGroups(log logr.Logger, cluster *fdbv1beta2.FoundationDBCluster, hasDesiredFaultTolerance bool) bool {
+func ReplaceFailedProcessGroups(log logr.Logger, cluster *fdbv1beta2.FoundationDBCluster, status *fdbv1beta2.FoundationDBStatus, hasDesiredFaultTolerance bool) bool {
```

Comment:
It's probably good enough to just pass down the `MaintenanceZone` instead of passing down the whole status struct?

Reply:
It is a pointer to the status struct. By propagating status we will have access to any other fields that we may need in this function (I think that's the way to look at it).